solver/src/main/java/org/chocosolver/solver/variables/view/IntView.java

```diff
@@ -167,29 +167,34 @@ public boolean removeValue(int value, ICause cause) throws ContradictionExceptio
     @Override
     public boolean removeValues(IntIterableSet values, ICause cause) throws ContradictionException {
         assert cause != null;
-        int olb = getLB();
-        int oub = getUB();
-        int nlb = values.nextValue(olb - 1);
-        int nub = values.previousValue(oub + 1);
-        if (nlb > oub || nub < olb) {
-            return false;
-        }
-        if (nlb == olb) {
-            // look for the new lb
-            do {
-                olb = nextValue(olb);
-                nlb = values.nextValue(olb - 1);
-            } while (olb < Integer.MAX_VALUE && oub < Integer.MAX_VALUE && nlb == olb);
-        }
-        if (nub == oub) {
-            // look for the new ub
-            do {
-                oub = previousValue(oub);
-                nub = values.previousValue(oub + 1);
-            } while (olb > Integer.MIN_VALUE && oub > Integer.MIN_VALUE && nub == oub);
-        }
-        // the new bounds are now known, delegate to the right method
-        boolean hasChanged = updateBounds(olb, oub, cause);
+        boolean hasChanged = false, fixpoint;
+        int nlb, nub;
+        do {
+            int olb = getLB();
+            int oub = getUB();
+            nlb = values.nextValue(olb - 1);
+            nub = values.previousValue(oub + 1);
+            if (!hasChanged && (nlb > oub || nub < olb)) {
+                return false;
+            }
+            if (nlb == olb) {
+                // look for the new lb
+                do {
+                    olb = nextValue(olb);
+                    nlb = values.nextValue(olb - 1);
+                } while (olb < Integer.MAX_VALUE && oub < Integer.MAX_VALUE && nlb == olb);
+            }
+            if (nub == oub) {
+                // look for the new ub
+                do {
+                    oub = previousValue(oub);
+                    nub = values.previousValue(oub + 1);
+                } while (olb > Integer.MIN_VALUE && oub > Integer.MIN_VALUE && nub == oub);
+            }
+            // the new bounds are now known, delegate to the right method
+            fixpoint = updateBounds(olb, oub, cause);
```

Comment:
I think that it comes from finishing conditions of the algorithm, but we are using "old" bounds variables her instead of new ones (nlb and nub). It might be clearer to call them.